roles/app_install/tasks/main.yml

```diff
@@ -74,7 +74,9 @@
   become_user: "{{ remote_admin_account }}"
   become: yes
   shell: |
-    source {{ app_py_virtual_env }}/bin/activate && pip3.8 install -r requirements/requirements.txt --no-index --find-links ./vendor/
+    source {{ app_py_virtual_env }}/bin/activate
+    python3 -m pip install --upgrade pip setuptools wheel
```

Comment:
Even as convention, I like this better. Splitting out commands unless there's a compelling reason to && them on one line.

Reply:
i agree and its easy to  understand and debug.